src/main/java/flight/reservation/order/FlightOrder.java

```diff
@@ -3,29 +3,45 @@
 import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import flight.reservation.Customer;
+import flight.reservation.Passenger;
 import flight.reservation.flight.ScheduledFlight;
 import flight.reservation.payment.CreditCard;
 import flight.reservation.payment.Paypal;
 
 public class FlightOrder extends Order {
     private final List<ScheduledFlight> flights;
-    static List<String> noFlyList = Arrays.asList("Peter", "Johannes");
+    private static List<String> noFlyList = Arrays.asList("Peter", "Johannes");
 
     public FlightOrder(List<ScheduledFlight> flights) {
         this.flights = flights;
     }
 
-    public static List<String> getNoFlyList() {
-        return noFlyList;
-    }
-
     public List<ScheduledFlight> getScheduledFlights() {
         return flights;
     }
 
-    private boolean isOrderValid(Customer customer, List<String> passengerNames, List<ScheduledFlight> flights) {
+    public static FlightOrder createOrder(Customer customer, List<String> passengerNames, List<ScheduledFlight> flights, double price){
+        if (!FlightOrder.isOrderValid(customer, passengerNames, flights)) {
+            throw new IllegalStateException("Order is not valid");
+        }
+        FlightOrder order = new FlightOrder(flights);
+        order.setCustomer(customer);
+        order.setPrice(price);
+        List<Passenger> passengers = passengerNames
+                .stream()
+                .map(Passenger::new)
+                .collect(Collectors.toList());
+        order.setPassengers(passengers);
+        order.getScheduledFlights().forEach(scheduledFlight -> scheduledFlight.addPassengers(passengers));
+        customer.addOrder(order);
```

Comment:
Adding the order to the customer might be a bit hidden and not clearly visible from the outside.